src/cil.mli

```diff
@@ -940,111 +942,113 @@ and label =
+    (** A [while(1)] loop. The termination test is implemented in the body of
      * a loop using a [Break] statement. If prepareCFG has been called,
      * the first stmt option will point to the stmt containing the continue
      * label for this loop and the second will point to the stmt containing
      * the break label for this loop. *)
 
-  | Block of block                      
-    (** Just a block of statements. Use it as a way to keep some block 
+  | Block of block
+    (** Just a block of statements. Use it as a way to keep some block
      * attributes local *)
 
-    (** On MSVC we support structured exception handling. This is what you 
-     * might expect. Control can get into the finally block either from the 
+    (** On MSVC we support structured exception handling. This is what you
+     * might expect. Control can get into the finally block either from the
      * end of the body block, or if an exception is thrown. *)
   | TryFinally of block * block * location
 
-    (** On MSVC we support structured exception handling. The try/except 
-     * statement is a bit tricky: 
-         [__try { blk } 
+    (** On MSVC we support structured exception handling. The try/except
+     * statement is a bit tricky:
+         [__try { blk }
          __except (e) {
             handler
          }]
 
-         The argument to __except  must be an expression. However, we keep a 
-         list of instructions AND an expression in case you need to make 
-         function calls. We'll print those as a comma expression. The control 
-         can get to the __except expression only if an exception is thrown. 
-         After that, depending on the value of the expression the control 
-         goes to the handler, propagates the exception, or retries the 
+         The argument to __except  must be an expression. However, we keep a
+         list of instructions AND an expression in case you need to make
+         function calls. We'll print those as a comma expression. The control
+         can get to the __except expression only if an exception is thrown.
+         After that, depending on the value of the expression the control
+         goes to the handler, propagates the exception, or retries the
          exception !!!
-     *)      
+     *)
   | TryExcept of block * (instr list * exp) * block * location
-  
 
-(** {b Instructions}. 
+
+(** {b Instructions}.
  An instruction {!Cil.instr} is a statement that has no local
 (intraprocedural) control flow. It can be either an assignment,
 function call, or an inline assembly instruction. *)
 
 (** Instructions. *)
 and instr =
-    Set        of lval * exp * location  
-   (** An assignment. The type of the expression is guaranteed to be the same 
+  Set        of lval * exp * location
+   (** An assignment. The type of the expression is guaranteed to be the same
     * with that of the lvalue *)
+  | VarDecl    of varinfo * location (** Generation of a VLA (i.e.
```

Comment:
`VarDecl` sounds too much like just Variable Declaration. Connection to VLA is not really clear except when reading the comment.

Reply:
Good point! This ties in with your comment over at https://github.com/goblint/analyzer/pull/74, I'll think about if we want to do this for all local declarations and then remove the comment, or change the name to something more specific.

Reply:
Now we have `Cil.instr += VarDecl` and `Goblint.MyCFG.edge += VDecl` where both are only used for VLAs.